flight/PiOS/Common/pios_streamfs.c

```diff
@@ -213,10 +213,22 @@ static int32_t streamfs_new_sector(struct streamfs_state *streamfs)
 	streamfs->active_file_arena_offset = 0;
 	streamfs->active_file_segment++;
 
-	if (streamfs_erase_arena(streamfs, streamfs->active_file_arena) != 0) {
+	// Test whether the sector has already been erased by checking the footer
```

Comment:
The purpose is to speed-up writing on a freshly formated file system. Instead of checking the whole page, it checks only the footer and erases the page if the footer isn't all `FF`.